### ISSUE TITLE
Checkout fields descriptions do not have corresponding aria-describedby

### ISSUE THREAD

**Issue by Shelob9** [easydigitaldownloads/easy-digital-downloads#5258]
Here is an example of markup for a field in my checkout form:

```
<p id="edd-email-wrap">
	<label class="edd-label" for="edd-email">
		Email Address									
		<span class="edd-required-indicator">*</span>
	</label>
	<span class="edd-description">We will send the purchase receipt to this address.</span>
	<input class="edd-input required" type="email" name="edd_email" placeholder="Email address" id="edd-email" value="">
</p>
```

If I'm reading [the spec](https://www.w3.org/TR/wai-aria/states_and_properties#aria-describedby) properly this is a bug. The correct markup would be something like this that adds an ID to the description and a reference to that element on the input by  aria-describedby:

```
<p id="edd-email-wrap">
	<label class="edd-label" for="edd-email">
		Email Address									
		<span class="edd-required-indicator">*</span>
	</label>
	<span class="edd-description" id="edd-email-description">We will send the purchase receipt to this address.</span>
	<input class="edd-input required" type="email" name="edd_email" placeholder="Email address" id="edd-email" value="" aria-describedby="edd-email-description">
</p>
```

Pretty sure all checkout fields with descriptions have this bug.

**Comment by pippinsplugins** [comment]
I don't know the spec very well.

@samikeijonen do you happen to know about this?

**Comment by samikeijonen** [comment]
I can take a look. At first look it feels a little bit weird that `<span class="edd-description" id="edd-email-description">We will send the purchase receipt to this address.</span>` is outside label. Screen readers won't announce text `We will send the purchase receipt to this address.` in this case.

And `aria-describedby` will help the screen readers announce that text. 

@Shelob9: Was that what you were thinking?

Anyways I can take a look more but my first reaction is to move it inside label like this:

```
<p id="edd-email-wrap">
	<label class="edd-label" for="edd-email">
		Email Address									
		<span class="edd-required-indicator">*</span>
               <span class="edd-description">We will send the purchase receipt to this address.</span>
	</label>
	<input class="edd-input required" type="email" name="edd_email" placeholder="Email address" id="edd-email" value="">
</p>
```

In required fields there probably should also be `aria-required="true" required` attributes.

**Comment by Shelob9** [comment]
This article on MDN suggestions adding an ID to the description and adding
an aria-describedby attribute referencing that ID to the element it
describes is correct.

developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute

**Comment by samikeijonen** [comment]
Yes it's correct but `<label>` is already doing description. Why not just stick with that. Or was there some other reason that I'm missing?

**Comment by Shelob9** [comment]
I think the distinction is that one is  a label and one is a description.
The label is semantically identified as a a label. The description looks
like a description on the screen, but there is no semantic markup
identifying it that way for screen readers.

On Nov 28, 2016 11:58 AM, "Sami Keijonen" <notifications@github.com> wrote:
>
> Yes it's correct but <label> is already doing description. Why not just
stick with that. Or was there some other reason that I'm missing?

**Comment by samikeijonen** [comment]
Got it. In that case I'm all for this change. Here is good summary of [how aria-describedby works](http://webaim.org/techniques/forms/advanced#describedby).

We should do full review what's happening on the front end.

**Comment by Shelob9** [comment]
@samikeijonen Are you going to WordCamp US? If so, id be happy to sit down and go over the accessibility markup issues in the EDDD forms with you. I've gotten pretty good at this stuff fixing similar bugs in Caldera Forms.

**Comment by samikeijonen** [comment]
@Shelob9 Sorry, I'm stuck in Finland:) But we can still take a closer look and open issues and PRs.
